### PR TITLE
ISPN-11298 Create TakeOfflineManager

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-11298

Moves the code that takes sites offline from BackupSender to TakeOfflineManager.

### CODE REVIEW

File: core/src/main/java/org/infinispan/xsite/BackupSenderImpl.java [hunk 1]
```diff
@@ -244,21 +173,18 @@ private void sendTo(XSiteReplicateCommand command, Collection<XSiteBackup> xSite
       if (rpcManager == null) {
          for (XSiteBackup backup : xSiteBackups) {
             XSiteResponse cs = transport.backupRemotely(backup, command);
+            takeOfflineManager.registerRequest(cs);
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/xsite/BackupSenderImpl.java]: Why do we no need to register the request when we didn't before? AFAICT, there was no call to `XSiteResponse#whenCompleted` previously

File: core/src/main/java/org/infinispan/xsite/status/TakeOfflineManager.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    * <p>
+    * If a site is online, then its value is {@link Boolean#TRUE}, otherwise is {@link Boolean#FALSE}.
+    *
+    * @return A {@link Map} with the site state.
+    */
+   Map<String, Boolean> status();
+
+   /*
+    * Brings a site with the given name back online.
+    */
```
[the remark this answers is not in the document]
Comment (pruivo) [reply core/src/main/java/org/infinispan/xsite/status/TakeOfflineManager.java]: yeah sorry!

File: core/src/main/java/org/infinispan/xsite/BackupSenderImpl.java (hunk 1 above)
Comment (pruivo) [reply core/src/main/java/org/infinispan/xsite/BackupSenderImpl.java]: the previous one is here
https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/xsite/BackupSenderImpl.java#L418

The old method updates the offline status and handles the exception. in this PR, it only handles the exception.

**Comment by ryanemerson** [comment]
Thanks @pruivo

**Pull request merged**